### PR TITLE
console: Configure gateway settings

### LINKED ISSUE #1868: Gateway general settings missing in Console
#### Summary

There are gateway fields that cannot be edited using the Console, e.g. `location_public`, `auto_update`

#### Why do we need this?

- Completeness
- With #1849 the `location_public` gateway field starts to have an important role. It should be easy for the user to view and edit.

#### What is already there? What do you see now?

I can edit other gateway settings

image

#### What is missing? What do you want to see?

Form fields for the missing gateway options

#### Environment

3.4.2

#### How do you propose to implement this?

Add missing fields.

#### Can you do this yourself and submit a Pull Request?

Most missing fields are boolean options. Can try to implement.

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #1868

#### Changes

- Add boolean fields for a few gateway settings

#### Notes for Reviewers

@johanstokking please review settings + descriptions
@bafonins @kschiffer console changes

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [X] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [X] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by neoaggelos** [comment]
Fixed CI

File: pkg/webui/console/components/gateway-data-form/index.js [hunk 1]
```diff
@@ -37,6 +37,13 @@ const m = defineMessages({
   gatewayDescPlaceholder: 'Description for my new gateway',
   gatewayDescDescription:
     'Optional gateway description; can also be used to save notes about the gateway',
+  public: 'Public',
```
Comment (bafonins) [inline pkg/webui/console/components/gateway-data-form/index.js]: To me it seems that `shared-messages` is a better place for this.

File: pkg/webui/lib/shared-messages.js [hunk 2]
```diff
@@ -106,12 +106,19 @@ export default defineMessages({
   frequencyPlan: 'Frequency Plan',
   fNwkSIntKey: 'FNwkSIntKey',
   nwkSKey: 'NwkSKey',
+  gatewayAutoUpdate: 'Automatic Updates',
```
Comment (bafonins) [inline pkg/webui/lib/shared-messages.js]:
```suggestion
  automaticUpdates: 'Automatic Updates',
```

similar changes for the rest of messages

File: pkg/webui/console/components/gateway-data-form/index.js [hunk 3]
```diff
@@ -117,13 +127,48 @@ class GatewayDataForm extends React.Component {
           name="gateway_server_address"
           component={Input}
         />
+        <Form.Field
+          title={sharedMessages.gatewayLocation}
+          name="location_public"
```
Comment (bafonins) [inline pkg/webui/console/components/gateway-data-form/index.js]: please add this to the validation schema of the form

```js
location_public: Yup.boolean().default(false)
```

File: pkg/webui/console/components/gateway-data-form/index.js [hunk 4]
```diff
@@ -117,13 +127,48 @@ class GatewayDataForm extends React.Component {
...
+          component={Checkbox}
+          label={sharedMessages.public}
+          description={m.locationDescription}
+        />
+        <Form.Field
+          title={sharedMessages.gatewayStatus}
+          name="status_public"
```
Comment (bafonins) [inline pkg/webui/console/components/gateway-data-form/index.js]: please add this to the validation schema of the form

```js
status_public: Yup.boolean().default(false)
```

File: pkg/webui/console/components/gateway-data-form/index.js [hunk 5]
```diff
@@ -117,13 +127,48 @@ class GatewayDataForm extends React.Component {
...
+          component={Checkbox}
+          label={sharedMessages.public}
+          description={m.statusDescription}
+        />
         <Message component="h4" content={sharedMessages.lorawanOptions} />
         <GsFrequencyPlansSelect name="frequency_plan_id" menuPlacement="top" required />
+        <Form.Field
+          title={sharedMessages.gatewayScheduleDownlinkLate}
+          name="schedule_downlink_late"
```
Comment (bafonins) [inline pkg/webui/console/components/gateway-data-form/index.js]: please add this to the validation schema of the form

```js
schedule_downlink_late: Yup.boolean().default(false)
```

File: pkg/webui/console/components/gateway-data-form/index.js [hunk 6]
```diff
@@ -117,13 +127,48 @@ class GatewayDataForm extends React.Component {
...
           title={m.dutyCycle}
           name="enforce_duty_cycle"
           component={Checkbox}
           label={m.enforced}
+          description={m.enforceDutyCycleDescription}
+        />
+        <Message component="h4" content={sharedMessages.gatewayUpdateOptions} />
+        <Form.Field
+          title={sharedMessages.automaticUpdates}
+          name="auto_update"
```
Comment (bafonins) [inline pkg/webui/console/components/gateway-data-form/index.js]: please add this to the validation schema of the form

```js
auto_update: Yup.boolean().default(false)
```

**Comment by neoaggelos** [comment]
Should I add #1840 while I'm at it, or do you think this is better suited for a separate PR (given that it needs a new component)?

**Comment by kschiffer** [comment]
Better to make a separate PR.

**Pull request merged**